syntax = "proto2";
package dtl.packets;


message ChildInfoRequest
{
    required int32 opcode = 1;
}

message ChildInfoResponse
{
    required int32 opcode = 1;
    required string name = 2;
    required int32 status = 3;    
    required string currentFn = 4;
    required bool hasGPU = 5;
}

message ChildComplete
{
    required int32 opcode = 1;
    required string name = 2;
}

message ChildSetInfo
{
    required int32 opcode = 1;
    required string name = 2;
}

message IssueTaskRequest
{
    required int32 opcode = 1;
    required string name = 2;
    required string function = 3;
    required bool needsGPU = 4;
    required bool hasParameters = 5;
}

message IssueTaskResponse
{
    required int32 opcode = 1;
    required string name = 2;
    required bool accepted = 3;
}

message TerminateAllChildrenRequest
{
    required int32 opcode = 1;
}

message TerminateAllChildrenResponse
{
    required int32 opcode = 1;
    required string name = 2;
}
